### PR TITLE
Add formation async status business logic implementation

### PR DESCRIPTION
**Pull request by PetarTodorovv**
**Description**
Add business logic for the formation async status API for `createFormation` and `deleteFormation` operations.

Changes proposed in this pull request:
- updating the formation state with the provided data through the API
- deleting formation as part of `deleteFormation` status update operation
- resynchronize formation notifications as part of `createFormation` status update operation
- unit tests adaptations
- adapt the e2e test to execute asynchronous notifications

**Related issue(s)**

- https://github.com/kyma-incubator/compass/pull/2886
- https://github.com/kyma-incubator/compass/pull/2928

**Pull Request status**

- [x] Implementation
- [x] Unit tests
- [ ] Integration tests
- [x] `chart/compass/values.yaml` is updated 
- [x] Mocks are regenerated, using the automated script

### CODE REVIEW

File: chart/compass/charts/gateway/templates/oathkeeper-rules.yaml [hunk 1]
```diff
@@ -438,7 +438,7 @@ spec:
   upstream:
     url: "http://compass-director.{{ .Release.Namespace }}.svc.cluster.local:{{ .Values.global.director.graphql.external.port }}"
   match:
-    methods: ["PATCH"]
+    methods: ["PATCH", "POST", "DELETE"]
```
Comment (dzahariev) [inline chart/compass/charts/gateway/templates/oathkeeper-rules.yaml:441]: PATCH seem to be unused now, should we remove it?

**Review by dzahariev — APPROVED** [review]

File: chart/compass/charts/gateway/templates/oathkeeper-rules.yaml (hunk 1 above)
Comment (PetarTodorovv) [reply chart/compass/charts/gateway/templates/oathkeeper-rules.yaml:441]: It is used for the formation assignment status API. There we use PATCH requests. The new ones - POST and DELETE are needed for the "new" formation status API

File: components/external-services-mock/internal/notification/handler.go [hunk 2]
```diff
@@ -383,128 +491,172 @@ func (h *Handler) asyncResponse(writer http.ResponseWriter, r *http.Request, ope
...
+		RequestBody: bodyBytes,
+	})
 
-func (h *Handler) ResetShouldFail(writer http.ResponseWriter, r *http.Request) {
-	h.shouldReturnError = true
-	writer.WriteHeader(http.StatusOK)
-}
+	formationIDFromBody := gjson.Get(string(bodyBytes), "details.id").String()
+	if formationIDFromBody == "" {
+		httputils.RespondWithError(ctx, writer, 500, errors.New("Missing formation ID from request body"))
```
[the remark this answers is not in the document]
Comment (PetarTodorovv) [reply components/external-services-mock/internal/notification/handler.go:672]: It was copy-paste from the old function. I will update it everywhere.

**Review by dzahariev — APPROVED** [review]

**Pull request merged**